mobile/test/java/io/envoyproxy/envoymobile/engine/testing/HttpTestServer.java

```diff
@@ -0,0 +1,55 @@
+package io.envoyproxy.envoymobile.engine.testing;
+
+import java.util.Collections;
+import java.util.Map;
+
+/** An HTTP test server. */
+public final class HttpTestServer {
+  /** The supported {@link HttpTestServer} types. */
+  public static class Type {
+    public static final int HTTP1_WITHOUT_TLS = 0;
+    public static final int HTTP2_WITH_TLS = 1;
+    public static final int HTTP3 = 2;
+    public static final int HTTP_PROXY = 3;
+    public static final int HTTPS_PROXY = 4;
+
+    private Type() {}
+  }
+
+  /** The instance of {@link HttpServer}. */
+  public static class HttpServer {
```

Comment:
Not sure if this class being nested is the best... HttpTestServer.HttpServer is a bit confusing in code. 

How about naming HttpTestServer to HttpTestServerProvider, and HttpServer to HttpTestServer?  Or something like that.  And both be top-level classes (i.e. no nesting).